internal/services/rewards.go

```diff
@@ -292,11 +296,40 @@ func (t *BaselineClient) assignPoints() error {
 			}
 		}
 
+		// TODO(ae): need to add signature here still, must update devicees-api grpc response to do so
+		earningVehicle, ok := attestationData[*ud.Vin]
+		if ok {
+			credExp, ok := earningVehicle[CredentialExpiresAt]
+			if ok {
+				if credExp.(time.Time).Before(ud.LatestVinCredential.Expiration.AsTime()) {
+					attestationData[*ud.Vin] = map[string]interface{}{
+						CredentialExpiresAt: ud.LatestVinCredential.Expiration.AsTime(),
+						Values:              []interface{}{*ud.TokenId, ud.DeviceDefinitionId, ud.LatestVinCredential.Id},
+					}
+				}
+
+			}
+		}
+		attestationData[*ud.Vin] = map[string]interface{}{
+			CredentialExpiresAt: ud.LatestVinCredential.Expiration.AsTime(),
+			Values:              []interface{}{*ud.TokenId, ud.DeviceDefinitionId, ud.LatestVinCredential.Id},
+		}
+
 		if err := thisWeek.Insert(ctx, t.TransferService.db.DBS().Writer, boil.Infer()); err != nil {
```

Comment:
I think the thing is we no longer know that every row in here is going to earn. I understand that we're not trying to implement _everything_ in this PR but I'm not sure where the responsibilities end and begin.